cherry_picker/cherry_picker/__init__.py

```diff
@@ -1,2 +1,2 @@
 """Backport CPython changes from master to maintenance branches."""
-__version__ = '1.2.1.dev1'
+__version__ = '1.2.1'
```

Comment:
btw what do you think of having scm-based versioning?

Reply:
What do you mean by scm-based versioning?

Reply:
Git tag being the source of truth for the version. Currently, it's manual update of version in Python module followed by tagging the commit. But there are flows, where it only sits in Git, so it would need you to just push a git tag to trigger the whole flow without extra steps.

Reply:
Ah ok. This version info is needed for `flit` though.

Reply:
Right, so when using setup.py I use setuptools_scm to make dist have info about version. It should be possible with flit as well.

Reply:
Sounds interesting?

Reply:
Hmm, I think I'll keep the current workflow for now.
Interesting idea though, I'll have to think about it. Thanks!

Reply:
Ok I'm interested now in deriving the version info automatically somehow, but I'd like to hear more first of what the workflow would look like.

Reply:
Well, I need to check what's available for flit, but in general it would be triggered just by pushing a tag and during build something figures out the version and puts it into metadata. setuptools_scm, for example, has a mode, where it'll create the python module from version if the project itself needs to use it (however it's also retrievable via pkg_resources part of setuptools).
In case of setuptools_scm, during development it uses last tag plus commit marker for versioning, but that's configurable.

As for prerequisites, you'll likely need to put a commit with changelog in git and tag afterwards.
Some ppl use tools like towncrier or reno to enforce changelog fragments be submitted along with each PR and then can fail tagged builds if those are not converted into common changelog.

I think I'll try submitting some demo PR with implementation once I have some time.

Reply:
Ok thanks. Before you go ahead with some demo PR, let me try to digest all of the above and decide whether it's worth the trouble. (sounds more complicated that I wanted) 😅 

I definitely want to further automate this somehow.

Reply:
Well, let's separate stages here:
1. Removing need to track version in files
2. Integrating changelog generator